### PR TITLE
thold sql optimization 

### PR DESCRIPTION
**Pull request by rgt0078**
Hello
Here is a means to optimize thold execution (see https://github.com/Cacti/plugin_thold/issues/126)
Regards

### CODE REVIEW

**Comment by ikorzha** [comment]
I have tested the changes you proposed on my prod system:
I get thousands of errors in the log see below:
thold-daemon

polling.php changes are fine, when I restore thold_functions.php original back no more errors

**Comment by rgt0078** [comment]
Sorry I fixed the cases where this function is called outside includes/polling.php (same behavior than before)

**Comment by ikorzha** [comment]
No more errors after latest commit, thank you. I do want to ask you, after I have applied the change I do not see any thold daemon improvements at all. I do not use standard thold poller as it is insufficient for the size of my install.
thold-daemon-poller

Any comments or improvements we can squeeze there too?

File: includes/polling.php [hunk 1]
```diff
@@ -403,33 +403,35 @@ function thold_check_all_thresholds() {
...
 				ON h.id = td.host_id
 				LEFT JOIN data_template_rrd AS dtr
 				ON dtr.id = td.data_template_rrd_id
 				WHERE td.thold_enabled = 'on'
 				AND (h.poller_id = 1 OR h.poller_id IS NULL)
-				AND td.tcheck = 1";
+				AND td.tcheck = 1 AND h.status=3";
 		} else {
-			$sql_query = "SELECT td.*, dtr.data_source_name
+			$sql_query = "SELECT td.*, dtr.data_source_name, h.hostname, h.description, h.notes, h.snmp_engine_id
```
Comment (cigamit) [inline includes/polling.php:416]: put the h.hostname ... on a second line.  Same with the other examples.  Indent with hard tabs.

File: includes/polling.php [hunk 2]
```diff
@@ -403,33 +403,35 @@ function thold_check_all_thresholds() {
...
 				FROM thold_data AS td
 				LEFT JOIN host AS h
 				ON h.id = td.host_id
 				LEFT JOIN data_template_rrd AS dtr
 				ON dtr.id = td.data_template_rrd_id
 				WHERE td.thold_enabled = 'on'
 				AND h.poller_id = " . $config['poller_id'] . "
-				AND td.tcheck = 1";
+				AND td.tcheck = 1 AND h.status=3";
```
Comment (cigamit) [inline includes/polling.php:424]: Put the AND ... on it's own line.

File: thold_functions.php [hunk 3]
```diff
@@ -1751,27 +1751,37 @@ function thold_check_threshold(&$thold_data) {
...
+			thold_debug('Threshold checking halted by Device Status (' . $h['status'] . ')' );
+			return;
+		}
 	}
 
 	/* ensure that Cacti will make of individual defined SNMP Engine IDs */
 	$overwrite['snmp_engine_id'] = $h['snmp_engine_id'];
 
 	/* pull the cached name, if not present, it means that the graph hasn't polled yet */
+	/* this request is useless !?!
```
Comment (cigamit) [inline thold_functions.php:1780]: Just remove it then.

**Review by cigamit — CHANGES_REQUESTED** [review]
Check my comments, make the changes, re-push, and thank you for taking the time.

**Comment by rgt0078** [comment]
Hi,
I inserted the same optimization in case of use of thold as a daemon...
Rgds

**Comment by ikorzha** [comment]
Thank you for your optimization, I have tested it with thold daemon. I do see a slight improvement on thold daemon now. Thank you for your improvements. Hopefully Cigamit will merge the changes in.

**Comment by cigamit** [comment]
Screwed up the merge, but it's fixed now.

**Pull request merged**